packages/postgres-database/src/simcore_postgres_database/utils_aiopg_orm.py

```diff
@@ -0,0 +1,201 @@
+
+        if PRIMARY_KEY in column_names:
+            # defaults to primery key
+            query = query.returning(self._primary_key)
+
+        elif ALL_COLUMNS in column_names:
+            query = query.returning(literal_column("*"))
+            is_scalar = False
+            # NOTE: returning = self._table would also work. less efficient?
+        else:
+            # selection
+            query = query.returning(*[self._table.c[name] for name in column_names])
+
+        return query, is_scalar
+
+    @staticmethod
+    def _check_access_rights(access: Set, values: Dict):
+        not_allowed: Set[str] = access.intersection(values.keys())
+        if not_allowed:
+            raise ValueError(f"Columns {not_allowed} are read-only")
+
+    @property
+    def columns(self) -> ImmutableColumnCollection:
+        return self._table.columns
+
+    def set_default(self, rowid: Optional[RowUId] = None, **unique_id) -> "BaseOrm":
+        """
+        Sets default for read operations either by passing a row identifier or a filter
+        """
+        if unique_id and rowid:
+            raise ValueError("Either identifier or unique condition but not both")
+
+        if rowid:
+            self._unique_match = self._primary_key == rowid
+        elif unique_id:
+            self._unique_match = functools.reduce(
+                operator.and_,
+                (
+                    operator.eq(self._table.columns[name], value)
+                    for name, value in unique_id.items()
+                ),
+            )
+        if not self.is_default_set():
+            raise ValueError(
+                "Either identifier or unique condition required. None provided"
+            )
+        return self
+
+    def clear_default(self) -> None:
+        self._unique_match = None
+
+    def is_default_set(self) -> bool:
+        # WARNING: self._unique_match can evaluate false. Keep explicit
+        return self._unique_match is not None
+
+    async def fetch(
+        self,
+        returning_cols: Union[str, List[str]] = ALL_COLUMNS,
+        *,
+        rowid: Optional[RowUId] = None,
+    ) -> Optional[RowProxy]:
+        query = self._compose_select_query(returning_cols)
+        if rowid:
+            # overrides pinned row
+            query = query.where(self._primary_key == rowid)
+        elif self.is_default_set():
+            assert self._unique_match is not None  # nosec
+            query = query.where(self._unique_match)
+
+        result: ResultProxy = await self._conn.execute(query)
+        row: Optional[RowProxy] = await result.first()
+        return row
+
+    async def fetchall(
+        self,
+        returning_cols: Union[str, List[str]] = ALL_COLUMNS,
+    ) -> List[RowProxy]:
+
+        query = self._compose_select_query(returning_cols)
+
+        result: ResultProxy = await self._conn.execute(query)
+        rows: List[RowProxy] = await result.fetchall()
+        return rows
+
+    async def update(
```

Comment:
what about adding an upsert option or method?

Reply:
yes, and also delete ... but i decided to add them as they are used.  For the moment only those are necessary. 
Next PR might include more if necesary